staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -174,117 +184,171 @@ func NewCmdApply(baseName string, f cmdutil.Factory, ioStreams genericclioptions
 		Long:                  applyLong,
 		Example:               applyExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			cmdutil.CheckErr(o.Complete(f, cmd))
+			o, err := flags.ToOptions(cmd, baseName, args)
+			cmdutil.CheckErr(o.Validate())
 			cmdutil.CheckErr(validateArgs(cmd, args))
```

Comment:
While you're at it, please move `validateArgs` and `validatePruneAll` to your newly created `Validate` function.